connectors/src/connectors/slack/bot.ts

```diff
@@ -294,14 +320,74 @@ export async function botValidateToolExecution(
     if (!slackChatBotMessage) {
       throw new Error("Missing Slack message");
     }
+    const slackClient = await getSlackClient(connector.id);
+
+    const userEmailHeader =
+      slackChatBotMessage.slackEmail !== "unknown"
+        ? slackChatBotMessage.slackEmail
+        : undefined;
+    let slackUserInfo: SlackUserInfo | null = null;
+    let requestedGroups: string[] | undefined = undefined;
+
+    if (slackUserId) {
+      try {
+        slackUserInfo = await getSlackUserInfoMemoized(
+          connector.id,
+          slackClient,
+          slackUserId
+        );
+      } catch (e) {
+        if (isSlackWebAPIPlatformError(e)) {
+          logger.error(
+            {
+              error: e,
+              connectorId: connector.id,
+              slackUserId,
+            },
+            "Failed to get slack user info"
+          );
+        }
+        throw e;
+      }
+    } else if (slackBotId) {
+      throw new Error("Unreachable: bot cannot validate tool execution.");
+    }
+
+    if (!slackUserInfo) {
+      throw new Error("Failed to get slack user info");
+    }
+
+    if (slackUserInfo.is_bot) {
+      throw new Error("Unreachable: bot cannot validate tool execution.");
+    }
+
+    const groups = await getRequestedGroupsForSlackUser(
+      connector,
+      slackClient,
+      slackUserInfo,
+      {
+        slackChannelId: slackChannel,
+        slackTeamId,
+        slackMessageTs,
+      },
+      slackConfig.whitelistedDomains
+    );
+    if (groups === null) {
+      return new Ok(undefined);
+    }
+
+    // If the user is allowed, we retrieve the groups he has access to.
+    requestedGroups = groups;
```

Comment:
Note that if you read the code of `notifyIfSlackUserIsNotAllowed` groupIds is [] always so the purpose of this function is to check the user is allowed and `isActiveMemberOfWorkspace` so the original name is WAY better after all and the requestedGroups logic is a no-op but can be kept for future use.